pkg/types/vsphere/validation/platform.go

```diff
@@ -60,30 +62,63 @@ func ValidateForProvisioning(p *vsphere.Platform, fldPath *field.Path) field.Err
 		allErrs = append(allErrs, field.Required(fldPath.Child("network"), "must specify the network"))
 	}
 
-	allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+	if len(p.APIVIP) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("apiVIP"), "must specify a VIP for API"))
+	}
+
+	if len(p.IngressVIP) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("ingressVIP"), "must specify a VIP for Ingress"))
+	}
 	return allErrs
 }
 
+// validateIPinCIDR checks if the given ip is within the machine CIDRs.
+func validateIPinCIDR(vipIP net.IP, machineNetwork []types.MachineNetworkEntry) bool {
```

Comment:
`validateIPinCIDR` -> `ipInNetwork`
nit: comment could be be slightly improved `ipInNetwork returns true if the given..."